torch/optim/optimizer.py

```diff
@@ -678,11 +683,11 @@ def register_load_state_dict_pre_hook(
     def register_load_state_dict_post_hook(
         self, hook: Callable[["Optimizer"], None], prepend: bool = False
     ) -> RemovableHandle:
-        r"""Register a load_state_dict post-hook which will be called after
-        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
+        r"""Register a state dict post-hook which will be called after
+        :meth:`~torch.optim.Optimizer.state_dict` is called. It should have the
         following signature::
 
-            hook(optimizer) -> None
+            hook(optimizer, state_dict) -> state_dict or None
```

Comment:
This is wrong..what inspires this change?

Reply:
It seems I mixed pre and post. Fixing them.